### DISCUSSION TITLE
Getting images from already loaded ImageLayer

### DISCUSSION THREAD

**Discussion by dmartin-gdms** [pelicanmapping/osgearth#1867] [Support Q&A] [answered]
I have loaded some ImageLayers and they are visible on the map. However, when I attempt to get various pieces of an image for some TileKey, it can take some time as it appears the driver wants to regenerate each image. Since the ImageLayer is already loaded and visible on the map, is it possible to get the image associated with some TileKey without regenerating that image? 

Below is an example of what I am currently doing:

```
std::vector<osgEarth::TileKey> tileKeys = getIntersectingTileKeys(extent);
osg::ref_ptr<osgEarth::ImageLayer> imageLayer = getImageLayer("layer_001");
osgEarth::ImageMosaic mosaic;
for(auto const& key : tileKeys) {
   //gdal driver regenerates the image for the tile key, despite already being visible on the map? 
   auto const image = imageLayer->createImage(key, nullptr); 
   if(image.valid()) {
      mosaic.getImages().emplace_back(image.getImage(), key);
   } 
}
...
```

**Comment by gwaldron** [comment, marked as answer]
No - Since the Map data model and the terrain engine are separate components, osgEarth does not have a way to access already-loaded tile images directly. It is on the "some day" list.

In the meantime, a cache might help with this.
